packages/modules/file/src/services/file-module-service.ts

```diff
@@ -49,6 +51,27 @@ export default class FileModuleService implements FileTypes.IFileModuleService {
     return Array.isArray(data) ? result : result[0]
   }
 
+  getUploadFileUrls(
+    data: GetUploadFileUrlDTO[],
+    sharedContext?: Context
+  ): Promise<UploadFileUrlDTO[]>
+  getUploadFileUrls(
+    data: GetUploadFileUrlDTO,
+    sharedContext?: Context
+  ): Promise<UploadFileUrlDTO>
+
+  async getUploadFileUrls(
+    data: GetUploadFileUrlDTO[] | GetUploadFileUrlDTO
+  ): Promise<UploadFileUrlDTO[] | UploadFileUrlDTO> {
+    const input = Array.isArray(data) ? data : [data]
+
+    const result = await Promise.all(
+      input.map((file) => this.fileProviderService_.getPresignedUploadUrl(file))
```

Comment:
suggestion: should we provide `getPresignedUploadUrls`?

Reply:
I just followed the same pattern as the `getPresignedDownloadUrl`, but since none of the providers provide this natively it's better to just handle the "concurrency" here